weave-js/src/components/Tag/Tag.tsx

```diff
@@ -100,11 +100,30 @@ export const RemovableTag: FC<RemovableTagProps> = ({
   Wrapper = Tailwind,
 }) => {
   const labelRef = useRef<HTMLParagraphElement>(null);
-  const isTooltipEnabled = isTagLabelTruncated(labelRef);
+  const [isTruncated, setIsTruncated] = useState(false);
+
+  useEffect(() => {
+    const checkTruncation = () => {
+      if (labelRef.current) {
+        setIsTruncated(isTagLabelTruncated(labelRef));
+      }
+    };
+
+    checkTruncation();
+    window.addEventListener('resize', checkTruncation);
+    
+    const timer = setTimeout(checkTruncation, 100);
+
+    return () => {
+      window.removeEventListener('resize', checkTruncation);
+      clearTimeout(timer);
+    };
+  }, [label]);
+
```

Comment:
Thank you so much for the feedback! I've removed the timer because it wasn't necessary, and removed the resize after further digging and finding that we do not adjust tag sizes based on our window sizes. 
<img width="1349" alt="image" src="https://github.com/user-attachments/assets/03163b78-bed3-405c-849a-44a7e08cde5c" />
<img width="495" alt="image" src="https://github.com/user-attachments/assets/616a074d-3f7d-4a76-a2cb-83aa8365f08f" />

same tag sizes  with different window sizes^